File: app/dbmodels/Supplier.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Supplier',
    'menu'          => 'Management'
);
$table['fields'] = array(
    'id'                 => DBModels::type('increments',
                                           array('visible' => false)),
    'name'               => DBModels::type('string'),
    'comment'            => DBModels::type('text'),
    'is_deleted'         => DBModels::type('boolean',
                                           array('visible' => false)),
    'supplier_type_id'    => DBModels::type('integer'),
    'address'            => DBModels::type('text'),
    'postcode'           => DBModels::type('string'),
    'town'               => DBModels::type('string'),
    'state'              => DBModels::type('string'),
    'country'            => DBModels::type('string'),
    'website'            => DBModels::type('string'),
    'phonenumber'        => DBModels::type('string'),
    'fax'                => DBModels::type('string'),
    'email'              => DBModels::type('string'),

);

$table['relationships'] = array(
    'supplier_type_id' => array(
        'type' => 'belongsTo',
        'item' => 'SupplierType'),
    'entity' => array(
        'type'  => 'belongsTo',
        'item'  => 'Entity'),
);
include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/supplier.py
from fusionglpi import db


class Supplier(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    supplier_type_id = db.Column(db.Integer, db.ForeignKey('supplier_type.id'))
    address = db.Column(db.Text)
    postcode = db.Column(db.String(255, None, True))
    town = db.Column(db.String(255, None, True))
    state = db.Column(db.String(255, None, True))
    country = db.Column(db.String(255, None, True))
    website = db.Column(db.String(255, None, True))
    phonenumber = db.Column(db.String(255, None, True))
    fax = db.Column(db.String(255, None, True))
    email = db.Column(db.String(255, None, True))

    infocoms = db.relationship('Infocom', backref='supplier')
